pkg/cmd/roachtest/roachstress.sh

```diff
@@ -54,24 +60,42 @@ fi
 mkdir -p "${a}"
 
 if [ ! -f "${cr}" ]; then
-  ./build/builder.sh mkrelease amd64-linux-gnu "build${short}"
-  mv "cockroach${short}-linux-2.6.32-gnu-amd64" "${cr}"
+  if [ -z "${LOCAL}" ]; then
+    ./build/builder.sh mkrelease amd64-linux-gnu "build${short}"
+    cp "cockroach${short}-linux-2.6.32-gnu-amd64" "${cr}"
+  else
+    make "build${short}"
+    cp "cockroach${short}" "${cr}"
+  fi
 fi
 
 if [ ! -f "${rt}" ]; then
-  ./build/builder.sh mkrelease amd64-linux-gnu bin/workload
-  mv -f bin.docker_amd64/workload "${wl}"
+  if [ -z "${LOCAL}" ]; then
+    ./build/builder.sh mkrelease amd64-linux-gnu bin/workload
+    cp bin.docker_amd64/workload "${wl}"
+  else
+    make bin/workload
+    cp bin/workload "${wl}"
```

Comment:
This copying can get a bit odd to follow, since `wl` sometimes evaluates to `bin/workload` although not when this branch is taken. Any reason why we don't simply set `$wl` to the built workload path and invoke/use it directly instead of copying it around?

Reply:
It's caching the binaries for a given SHA. I've found that this saves me a lot of time, because I often jump around between branches juggling multiple repro attempts, and the builder vs direct-make invocations also nuke each other's caching somehow, so it really pays off if you only build stuff at a given SHA once.

I am confused though, when does `wl` evalaute to `bin/workload`? `wl` and the other two-digit short-hands are always in the artifacts directory. Am I missing something?

Reply:
Ah, my bad.